File: go/raindrops/raindrops.go
/*
Package raindrops is a spin on the classic FizzBuzz problem
*/
package raindrops

import "strconv"

// Convert takes in an int n and either returns n as a string or
// returns a rain sound depending on its factors
func Convert(n int) string {
	result := ""
	if n%3 == 0 {
		result += "Pling"
	}
	if n%5 == 0 {
		result += "Plang"
	}
	if n%7 == 0 {
		result += "Plong"
	}
	if len(result) == 0 {
		result = strconv.FormatInt(int64(n), 10)
	}

	return result
}

File: typescript/raindrops/raindrops.ts
const rainSounds = new Map<number, string>([
  [3, "Pling"],
  [5, "Plang"],
  [7, "Plong"],
])

/**
* @param value - [TODO:description]
* @returns [TODO:return]
*/
export function convert(value: number): string {
  let result = ''
  rainSounds.forEach((sound: string, operand: number) => {
    if (value % operand == 0) {
      result = `${result}${sound}`
    }
  })
  return result.length > 0 ? result : `${value}`
}
